src/components/tools/Location/StaticMap.svelte

```diff
@@ -34,84 +35,84 @@
-      // Padding cannot be used without the auto parameter/bounding box.
-      src = `https://api.mapbox.com/styles/v1/${style}/static/geojson(${overlay})/${bounds}/${width}x${height}?&access_token=${mapboxgl.accessToken}`;
-    } else {
-      // Add style for non point geometry
-      geojson = {
-        type: "Feature",
-        properties: {
-          "fill-opacity": 0,
-          "stroke-width": 3,
-        },
-        geometry: location.geometry,
-      };
-      // The Static Images API only accepts requests that are 8,192 or fewer characters long.
-      // Simplify geometry only for more complex shapes
-      const geojsonString = JSON.stringify(geojson);
-      if (geojsonString.length < 7000) {
-        overlay = encodeURIComponent(geojsonString);
-      } else {
-        const simplifiedGeojson = simplify(geojson, {
-          tolerance: 0.01,
-          highQuality: true,
-        });
-        overlay = encodeURIComponent(JSON.stringify(simplifiedGeojson));
-      }
-      bounds = "auto";
+  function createSrcUrl({ overlay, bounds, params }) {
+    return `https://api.mapbox.com/styles/v1/${style}/static/geojson(${overlay})/${bounds}/${width}x${height}?${serialize(
+      params
+    )}`;
+  }
 
-      src = `https://api.mapbox.com/styles/v1/${style}/static/geojson(${overlay})/${bounds}/${width}x${height}?padding=${padding}&access_token=${mapboxgl.accessToken}`;
+  function createOverlay(geojson, tolerance = 0.005) {
+    const overlay = encodeURIComponent(JSON.stringify(geojson));
+    if (overlay.length < 11000) {
+      return overlay;
     }
+    const simplifiedGeojson = simplify(geojson, {
+      tolerance,
+      highQuality: true,
+    });
+    return createOverlay(simplifiedGeojson, tolerance + 0.05);
+  }
+
+  function getPointImgSrc({ geometry, center }) {
+    // Set bounds to use zoom instead of auto
+    // This prevents static image from being zoomed in too much
+    // Note: Padding cannot be used in conjunction with zoom.
+    const bounds = `${center[0]},${center[1]},${zoom}`;
+    const overlay = createOverlay(geometry);
+    const params = { access_token: accessToken };
+    return createSrcUrl({ overlay, bounds, params });
   }
 
+  function getPolygonImgSrc({ geometry }) {
+    // Add style for non point geometry
+    const geojson = {
+      type: "Feature",
+      properties: {
+        "fill-opacity": 0,
+        "stroke-width": 3,
+      },
+      geometry,
+    };
+    const bounds = "auto";
+    const params = { access_token: accessToken, padding };
+    // The Mapbox Static Images API only accepts requests that are 8,192 or fewer characters long.
+    // Truncate coordinate precision
+    const truncatedGeojson = truncate(geojson, { precision: 4 });
+    // Simplify geometry
+    const overlay = createOverlay(truncatedGeojson);
+    return createSrcUrl({ overlay, bounds, params });
+  }
+
+  function handleLocation(feature) {
+    alt = `map of ${feature.title}`;
```

Comment:
I personally wouldn't set the value of `alt` here and might instead make it a reactive variable like 

```js
$: alt = location ? location.title : ""
```

An empty string (`""`) is a valid `alt` text attribute and indicates to screen readers to ignore the image. 

I think you're also using both a local variable and global variable for the same thing within this function. It would make it easier to follow to just use the local variable `feature` IMO.

Reply:
Fixed via https://github.com/berkeley-gif/caladapt-website-2021/pull/74/commits/c9e86d7a69f548af26546dedbdca05b9676b944f